### PR TITLE
Add SEP-38 tests

### LINKED ISSUE #71: Add tests for SEP-38 and related functionality
We should offer a set of tests for testing the SEP-38 API as well as the SEP-6 and SEP-31 requests that use quotes.

### PR DESCRIPTION
**Pull request by JakeUrban**
resolves #71

Adds tests to the test suite library for SEP-38.

One major assumption made by these tests is that the anchor supports clients looking to sell Stellar assets in exchange for off-chain assets, which are the types of quotes used to facilitate SEP-6/24 withdrawals and SEP-31 sends. If the anchor doesn't support these types of quotes the suite will not be very useful to them. Ideally we would check which type of exchange is supported and use those types of quotes in the rest of the tests.

### CODE REVIEW

File: @stellar/anchor-tests/src/helpers/console.ts [hunk 1]
```diff
@@ -76,15 +76,6 @@ async function printColoredTextTestRun(testRun: TestRun, verbose: boolean) {
...
   }
-  if (verbose && testRun.result.failure && testRun.result.failure.links) {
-    console.log(c.bold("Relevant Links:\n"));
-    console.group(); // failure links group
-    for (const linkLabel in testRun.result.failure.links) {
-      console.log(`${linkLabel}: ${testRun.result.failure.links[linkLabel]}`);
-    }
-    console.groupEnd(); // failure links group
-    console.log();
-  }
```
Comment (piyalbasu) [inline @stellar/anchor-tests/src/helpers/console.ts:87]: just curious why this is being removed in this PR. With my limited context, it looks unrelated to SEP-38 tests

**Review by piyalbasu — APPROVED** [review]

File: @stellar/anchor-tests/src/helpers/console.ts (hunk 1 above)
Comment (JakeUrban) [reply @stellar/anchor-tests/src/helpers/console.ts:87]: Its just duplicate logic, the section above already outputs the links. I noticed when running via the CLI that the links were outputted twice when using verbose.
